src/Model/ChannelPricingLogEntry.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\PriceHistoryPlugin\Model;
+
+use Sylius\Component\Core\Model\ChannelPricingInterface;
+
+class ChannelPricingLogEntry implements ChannelPricingLogEntryInterface
+{
+    private ?int $id = null;
+
+    private \DateTimeInterface $loggedAt;
+
+    public function __construct(
+        private ChannelPricingInterface $channelPricing,
+        private int $price,
+        private ?int $originalPrice,
+    ) {
+        $this->loggedAt = new \DateTimeImmutable();
+    }
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getChannelPricing(): ChannelPricingInterface
+    {
+        return $this->channelPricing;
+    }
+
+    public function getPrice(): int
```

Comment:
if I am correct, it is possible for this price to be null, in the case that we define prices for a product that is not enabled in a given channel

Reply:
Oh, I didn't notice it. However, IMO null prices should not be logged. I mean, if the channel is disabled it is not a price change in the catalog. If the admin decides to reenable some channels, he will have to specify the price back – then the system will log this change.

Reply:
Oh, this is an interesting case, which I don't think we have discussed before. It makes sense because otherwise, we could calculate this lowest price of 30 days incorrectly. Nevertheless, this should be covered in the scenarios, or even best, a separate task should be created for this, so that in this one the focus would already be on the logging and calculation itself, without additional conditions.

Reply:
Alternatively, we should always log the change but mark whether the product was on sale (enabled and enabled in channel) at the time. Then we will always have the full history, because this plugin, although mainly intended to solve the case of the omnibus, is also intended to provide the functionality of price history. 
I've pinged @CoderMaggie to create a proper task to resolve that case.